### PR TITLE
fix: calls crash on simulator video [WPB-8954]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8954" title="WPB-8954" target="_blank">BugWPB-8954</a>  iOS: calls crash on iOS 17 simulator
  </td></table>
  <br />
 

### Issue

The app crashes on iOS 17 simulators.

The solution creates a video container as simple view, that is added initially. Only when the video setup is required, the real `AVSVideoView` is created and embedded.

So then it's possible to do audio only calls, but video calls still crash when the video rendering activates.

### Testing

- Tests call between iOS 16 and 17 simulators
- Test real devices and simulators
- Test real between devices
- Start calls with audio only
- Start calls with video
- Receive calls with audio only
- Receive calls with video

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/CallParticipantView.swift [hunk 1]
```diff
@@ -161,4 +157,29 @@ final class CallParticipantView: BaseCallParticipantView {
...
+    // MARK: Override Base
+
+    override func updateVideoShouldFill(_ shouldFill: Bool) {
+        if shouldFill, videoView == nil {
+            // [WPB-8954] Setup video only when the video really starts to avoid
+            // calls crashing on the iOS 17 simulator.
+            let videoView = makeVideoView()
+            self.videoView = videoView
+
+            videoContainerView?.setupVideoView(videoView)
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/CallParticipantView.swift:170]: So will this still crash when video is enabled?

Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/CallParticipantView.swift:170]: Unfortunately yes. It just makes it possible to have audio only calls in the simulator again (as on iOS 16 simulators it was working).

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/AVSVideoContainerView.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import UIKit
+import struct WireSystem.WireLogger
+
+/// A placeholder container for AVSVideo to start the rendering only if the view is instantiated and setup.
+final class AVSVideoContainerView: UIView {
+
+    private weak var videoView: UIView?
+
+    init() {
+        super.init(frame: .zero)
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/AVSVideoContainerView.swift]: I think the default implementation of `init()` does exactly the same.

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/AVSVideoContainerView.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/AVSVideoContainerView.swift]: Maybe both inits can be removed.

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/BaseCallParticipantView.swift [hunk 4]
```diff
@@ -224,7 +224,11 @@ class BaseCallParticipantView: OrientableView, AVSIdentifierProvider {
     private func updateFillMode() {
         // Reset scale if the view was zoomed in
         scalableView?.resetScale()
-        videoView?.shouldFill = shouldFill
+        updateVideoShouldFill(shouldFill)
+    }
+
+    func updateVideoShouldFill(_ shouldFill: Bool) {
+        // TODO: override in subsclasses
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/BaseCallParticipantView.swift]:
```suggestion
        fatalError("override in subsclass")
```

**Review by caldrian — APPROVED** [review]
👏

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/BaseCallParticipantView.swift (hunk 4 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/BaseCallParticipantView.swift]: I had this before, but some unit tests uses this base view through several layers 🤷‍♂️ So I turned it into a comment..

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**